Script changes for tensorflow

ethantang95 opened the pull request.

TimZaman committed:
Basic Tensorflow Support

Added some initial tf tools

Implemented UI

Fixes for tensorflow 0.10

Removed tf-slim as its not part of the 0.10 master

Added the lmdb reader with a tf.cond that needs replacement

Implemented train and val seperation with a templating

Fixed issue with dequeueing both runners by pulling both graphs

Implemented training and validation rythm

Added support for both png and jpg and added 16 bit support

Implemented mean subtraction - but needs rework to load as constant

Added an optimized implementation of mean subtraction

Further optimized the mean loading by using a shared constant

Wrapped the data loader in a factory to easily support more data types

Implemented cropping

Implemented floating point support. Implemented seperate LMDB database. Implemented regression support. Added some brief nosetests. Need to invoke accuracy only on classification though.

Implemented variable restoration. Needs thorough testing

Implemented inferencing, not entirely polished

Moved some code into functions, started on modularization a bit

Implemented digits custom helper functions

Implemented custom printing ops

Implemented autoencoder

total rewrite of summaries

Implemented output to console from scalar summaries

Fixes for summary outputs: only simple scalar values are parsed to console

Implemented binary segmentation and necessary fixes

Some updates on binary seg

Implemented all possible optimizers and started work on learning rate shaper

Started work on the lr policies

Fixes for learning_rates, implemented optimizers, tested variable summary output to UI

Implemented and tested all learning rates and optimizers

Introduces new model definition and improvements in loss handling and graph layout

Major refactoring of main code. Implemented new model description. Implemented and tested inferencing. Implemented and tested weight/snapshot loading.

All-round minor updates and fixes

Fixes in summary cumulator and implemented an RNN model

Fixes for mean subtraction in tf and tf-ui, implemented data order selection in image-view extension

Implemented support for mean file of format: png, jpg, binaryproto - the latter being the fault that DIGITS will provide.

Added support for runtime statistics and some allround fixes

Added static tensorboard style network visualization for tensorflow. Added output of traces (no vis yet). Added a loader while waiting for network vis. Minor syntax cleanups.

Implemented alexnet standard network

Pulled in updates for travis build and added tensorflow install

Added two more files for Mr Travis

Implemented tensorflow configuration

Added tf config to doc

Fixes for ubuntu deployment of tf.

Moved tf tools

Fixes for tf ubuntu

Fixes for tf ubuntu

Some fixes and updates for TF in Travis

Fix in network viz test

Implemented default sinlge-gpu support and some nosetests

Fixes for inference

Added siamese network, bugfixes, minor features, some utility tf functions

Added siamese network and example png

Better error-ui format for network viz

Added an alternative simpler siamese network that doesnt need a seperate db, minor error update

Preliminary version of hdf5 implemented

Implemented fine-tuning by renaming variables

Implemented visualisation of variables and the activations of the Ops they belong to.

Fix in inf vis naming

Fixes in visualualisation shapes and naming

Implemented softmax upon classification

Implemented all nosetests for tf classification, and many allround bugfixes

Implemented generic nosetests - some need work

Fix for travis to find python exe

Implemented a better file format deducer, and implemented a bare minimal TFRecord-reader

Added top_n accuracy shortcut

Implemented on-line data augmentation for TF, 5 types. Some minor bugfixes. Need to do something with image whitening though during validation and inf..

Added tensorflow data augmentation test

Minor fixes and improvements from linter

Implemented minimal and bare multigpu and fixes to get it running for greg

Preliminary version of tfrecord writer for classification

Some changes to optimize dataloading for tfr

More fixes for tfrecrods

Fix generic data loading

Minor breaking changes but updates in namescoping

Implemented new model structure. Improvements to multi-gpu handling. Updates to namespaces. Implemented accounting for regularization. Many allround updates

Implemented proper visualisation for gpu devices

Minor updates and converted alexnet and vgg16 to new format

Fix in tfrecord shape

WIP on timeline traces

Finalized support for tensorflow timeline traces

Fixed alexnet for tf

Fix merge errors

Minify tf-graph-basic.build.js

gheinrich committed:
bAbI data plug-in

Add utils

Add inference form to bAbI dataset

Allow inference without answer

Allow unknown words in BaBI data plug-in

Fix bAbI plugin Lint errors

gheinrich committed:
Tensorflow integration updates

Use TFRecords for TF inference

TF: Don't rescale inputs

Fix some TF classification tests

Remove unnecessary print

Fix TF imports when uninstalled

Fix mean image scale

Fix generic model tests

Fix Torch single image inference

Fix inference

TMP TF Lint

Revert changes in digits-lint script

Lint: ignore tensorflow standard examples

More Lint fixes

gheinrich committed:
Add gradient hook

gheinrich committed:
Add memn2n model

gheinrich committed:
Update memn2n with gradient hooks

gheinrich committed:
GAN example

gheinrich committed:
Make batch size variable

gheinrich committed:
Training/inference paths

gheinrich committed:
Small update to TF 0.12

gheinrich committed:
Snapshot names, float inference, restore all vars

gheinrich committed:
Add TB link

gheinrich committed:
Do not restore global_step or optimizer variables

gheinrich committed:
Update GAN network

gheinrich committed:
Dynamically select inference form

gheinrich committed:
TF inference: convert images to float

gheinrich committed:
Update GAN z-gen network

gheinrich committed:
Small Update model view layout

gheinrich committed:
Add GAN plug-ins

gheinrich committed:
Update GAN plug-in to create CelebA dataset

gheinrich committed:
Add ability to show input in ImageOutput extension

gheinrich committed:
Add all data to raw data view extension

gheinrich committed:
Add model for CelebA dataset

gheinrich committed:
Update GAN data plug-in

gheinrich committed:
Update all losses in one session

gheinrich committed:
Remove conversion to .png in GAN data plug-in

gheinrich committed:
TF Slim Lenet example

Divide input by 255

gheinrich committed:
Update GAN data plug-in

gheinrich committed:
Fix TF model snapshot

gheinrich committed:
Reduce scheduler delays to speed up inference

gheinrich committed:
Update GAN plugins

gheinrich committed:
Fix TF tests

gheinrich committed:
Add API to LmdbReader (used by gan_features.py)

gheinrich committed:
Add GAN walk-through

gheinrich committed:
Save animated gif

gheinrich committed:
Update GAN walkthrough with embeddings video

gheinrich committed:
Fix GAN view for list encoding

gheinrich committed:
Add animation task to GAN plugins

gheinrich committed:
Add view task to see image attributes

gheinrich committed:
Add comments to GAN models

gheinrich committed:
Update README

gheinrich committed:
Fix GAN features script

gheinrich committed:
GAN app

gheinrich committed:
Fix DIGITS inference

gheinrich committed:
Adjust GAN window size automatically

gheinrich committed:
Add attributes to GAN app

gheinrich committed:
Move gandisplay.py

gheinrich committed:
Remove wxpython 3.0 selection

gheinrich committed:
Fix call to model

gheinrich committed:
Adding disclaimer

ethantang95 committed:
Ported DIGITS to using tensorflow 1.1.0.
Got master branch working

ethantang95 committed:
Ported DIGITS to using tensorflow 1.1.0.

gheinrich committed:
Merge pull request #1 from ethantang95/TF-Work

Worked on porting tensorflow to version 1.1. All unit test passes

ethantang95 committed:
first cherrypick for installation scripts

ethantang95 committed:
updated gitignore

ethantang95 committed:
Merge branch 'master' into TF-Remote-Work

ethantang95 commented:
@lukeyeager

lukeyeager commented on .travis.yml:
```diff
@@ -126,7 +127,6 @@ before_install:
 install:
     - mkdir -p ~/.config/matplotlib
     - echo "backend:agg" > ~/.config/matplotlib/matplotlibrc
-    - ./scripts/travis/install-openblas.sh $OPENBLAS_ROOT
```
Read the commit message which added `install-openblas.sh`: https://github.com/NVIDIA/DIGITS/commit/e6be9fef83eb18419c836c116c9a41956928b610

If you stop building openblas locally, then you'll need to address this issue with Torch. Either don't cache the torch build, force Torch to stop building openblas from source, or find a way to re-install the openblas build which was cached along with the Torch build.

ethantang95 replied:
The problem we are seeing though is that building openblas locally breaks something in numpy which is required by tensorflow. Maybe we can only build openblas in the torch build and leave tensorflow alone?

ethantang95 replied:
openblas is only for torch right? or do we need it for caffe/TF also?

lukeyeager replied:
Maybe try building a different version of OpenBLAS?

A BLAS library is going to be required for all the frameworks. Since Travis doesn't have GPUs, we can't use cuBLAS, etc. You need ATLAS or OpenBLAS.

lukeyeager commented on scripts/travis/install-tensorflow.sh:
```diff
@@ -0,0 +1,16 @@
+LOCAL_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+if [ ! -z "$DEB_BUILD" ]; then
+    echo "Skipping for deb build"
```
Not necessary since https://github.com/NVIDIA/DIGITS/commit/543ebb6340b1b656075fcd3a835af4b315961e3b

lukeyeager commented on scripts/travis/install-torch.sh:
```diff
@@ -29,8 +29,8 @@ cd "$INSTALL_DIR"
-LOG_FILE="$LOCAL_DIR/torch-install.log"
-./install.sh -b >"$LOG_FILE" 2>&1 || (cat "$LOG_FILE" && false)
+LOG_FILE=$LOCAL_DIR/torch-install.log
+{ ./install-deps && ./install.sh -b ; } >$LOG_FILE 2>&1 || (cat $LOG_FILE && false)
```
Why is this helpful?

ethantang95 committed:
Tf install experimental (#2)

* Fix visualization when palette is None (#1177)

The palette may be `None`when working with grayscale labels.
Fix #1147

* Bugfix for customizing previous models (#1202)

* [Packaging] Disable tests (#1227)

* [Tests] Skip if extension not installed (#1263)

* [Docs] Fix spelling errors in comments

* [Docs] Add note about torch pkg and cusparse (#1303)

* [Docs] Add note about torch pkg and cusparse (#1303)

* [Caffe] Fix batch accumulation bug (#1307)

* Use official NVIDIA model store by default (#1308)

* Mark v5.0.0

* [Packaging] Pull latest docker image before build

* Add .pgm to list of supported image file formats

* Restrict usage of cmap to labels DB in generic dataset exploration

fix #1322

* Update Object Detection example doc (#1323)

* Update Object Detection example doc (#1323)

* [TravisCI] Cache local OpenBLAS build

This fixes a Torch bug we've been having on Travis for a while now.

We had only been building OpenBLAS from source when there was no cached
torch build present on the build machine. That meant you could get a
cached build of Torch which was built against one version of OpenBLAS,
but the system actually installed an older version. This led to memory
corruption and segmentation faults.

* [Tests] Skip if extension not installed (part 2) (#1337)

* [TravisCI] Install all plugins by default

Also test no plugins

* [Tests] Skip if extension not installed (#1337)

* [Docs] Update model store documentation (#1346)

TODO: add a screenshot of the official model store once approved

* [Docs] Update model store documentation (#1346)

TODO: add a screenshot of the official model store once approved

* Add steps to specify the Python layer file (#1347)

* Add steps to specify the Python layer file (#1347)

* [Docs] Install minimal boost libs for caffe

* Remove the selenium walkthrough

* Update copyright year for 2017

* Add a few missing copyright notices

* Fix Siamese example

Broadcast -1 into all elements that equal 0 in original label.

* Fix Siamese example (#1405)

Broadcast -1 into all elements that equal 0 in original label.

* [Packaging] Make nginx site easier to customize

* Fix documentation typo. train.txt and test.txt was swapped and shown in the wrong folders for mnist and cifar10 data sets.

* Document a cuDNN workaround for text example (#1422)

* Document a cuDNN workaround for text example (#1422)

* Correct shebang for prepare_pascal_voc_data.sh (#1450)

* [Docs] Document workaround for torch+hdf5 error

* Fix typo in ModelStore.md

* Fix typo in medical-imaging/README.md

* Fix bash lint with shellcheck

* Fix bugs when visiting nested image folder

* Fix shellcheck-related bug in PPA upload script

* Copy labels.txt inside the dataset

Move import to the top

* Fix Distribution Graph

Move backwards-compatibility to setstate

* Fix typo in Sunnybrook plug-in

* Fix a bug introduced when fixing shellcheck lint

* Fix another shellcheck-related bug

* Fix table formatting in README.md

Fix table formatting

* Clamp distance values from segementation boundaries before begin converted to uint8. That was causing banding in the image because of wrapping at V % 256

* lint

* [Docs] 5.0 debs and Ubuntu 16.04 support

* WIP lint fix

* Linted most of what I can lint prior to asking for context

* updated the model store urls in the readme

* added debugs in build scripts to understand the point of failure

* added travis wait to install openblas

* removed tensorflow to the build process to see if affects openblas

* removed suppressing log contents

* added set -x

* fixed control

* re-enabling tensorflow to see if travis builds

* updated the version of numpy to ensure a stable build for travis wrt to open issue 8653 on numpy github

* forcing numpy to v 1.8.1

* added the official store image and updated the documentation (#1650)

* [TravisCI] Add `git fetch --unshallow` for DIST

Useful for TravisCI builds in forks.

* Got travis script to work for tensorflow installation

ethantang95 committed:
merge with greg master

ethantang95 committed:
merged with origin

ethantang95 committed:
Cleaning installation to work with Numpy 1.3 upgrade

removed the open blas stuff that somehow made it into here

embarassing merge residue

force install specific numpy version because 1.13 was being installed

asdf

trying changing the tensorflow install

reodered the installation order to see if it builds due to TF using
numpy 1.13 now

ethantang95 committed:
Tf example (#3)

* inital work on autoencoder TF example

* Moved the example files to its proper location

* atempting to get autoencoder to work

* autoencoder work

* validated tensorflow autoencoder example

* updated gitignore

* disabled comments in the segmentation-model.lua script to prevent crashing

* commiting the changes made to binary segmentation tf

* adding work to do something else

* I am seriously wayy too tired to write this commit message, it's just random bits of stuff

* got binary seg and siamese working

* started to work on the regression network

* milestone

* got regression for TF working

* Got fine tuning to work in TF

* changed the code to the format that is wanted by tim and greg

* Finished all the work for examples

inital work on autoencoder TF example

Moved the example files to its proper location

atempting to get autoencoder to work

autoencoder work

validated tensorflow autoencoder example

updated gitignore

disabled comments in the segmentation-model.lua script to prevent
crashing

commiting the changes made to binary segmentation tf

adding work to do something else

I am seriously wayy too tired to write this commit message, it's just
random bits of stuff

got binary seg and siamese working

rebase

rebase

started to work on the regression network

milestone

got regression for TF working

Got fine tuning to work in TF

changed the code to the format that is wanted by tim and greg

got fine tuning working

* Some small fixes

* changes WRT PR

trying renaming the weights

tested renaming variables

ethantang95 committed:
fixed api problem for multi gpus

ethantang95 committed:
merged with my work master branch

ethantang95 committed:
git removed installing tests

ethantang95 committed:
updated most of linting

ethantang95 committed:
Removed unused block of code as per suggestion by Greg

ethantang95 committed:
Removing spaces...
